src/rest/utils.js

```diff
@@ -3,16 +3,18 @@ import debugFactory from 'debug'
 
 const debug = debugFactory('StreamrClient:utils')
 
-export const authFetch = async (url, apiKey, opts = {}) => {
+export const authFetch = async (url, session, opts = {}, requireNewToken = false) => {
     debug('authFetch: ', url, opts)
+    const newHeaders = opts.headers ? opts.headers : {}
+    if (session) {
+        const token = await session.getSessionToken(requireNewToken)
+        newHeaders.Authorization = `Bearer ${token}`
```

Comment:
Here you're modifying the `opts.headers` object which is potentially passed from outside this function, which is bad practice as the function effects leak outside the function scope. One idea would be to do something like (untested code):

```
const authHeader = {}
if (session) {
    const token = await session.getSessionToken(requireNewToken)
    authHeader.Authorization = `Bearer ${token}`
}

const req = {
    ...opts,
    headers: {
        ...authHeader,
        ...(opts.headers || {})
    }
}
```